sympy/core/symbol.py

```diff
@@ -226,9 +226,20 @@ def _diff_wrt(self):
         """
         return True
 
+    @staticmethod
+    def _validate_assumptions(assumptions):
+        """Validate assumption strings.
+        """
+        valid_assumptions = set(_assume_defined)
+        diff = set(assumptions) - valid_assumptions
+        if diff:
+            raise ValueError(
```

Comment:
As noted in the discussion, it would be better to make this a warning (`warnings.warn(...)`) instead of an error. That way the existing behavior continues to work for those using it on purpose, but people will be alerted to typos. The warning message should also be something like "unknown assumptions ..." instead of "invalid assumptions ...".